yt_dlp/extractor/vueschool.py

```diff
@@ -0,0 +1,151 @@
+from .common import InfoExtractor
+from .vimeo import VimeoIE
+from ..utils import (
+    int_or_none,
+    get_element_by_class,
+    get_element_text_and_html_by_tag,
+    extract_attributes,
+    compat_HTMLParseError
+)
+
+
+class VueSchoolBaseIE(InfoExtractor):
+    def _get_course_info_from_lesson_slug(self, lesson_slug, id):
+        return self._download_json(f"https://vueschool.io/api/lessons/{lesson_slug}/widgetInfo", video_id=id)
+
+    def __flatten_list_elements(self, lists):
+        output = []
+        for list in lists:
+            output += list
+        return output
+
+    def _flat_lessons_from_chapters(self, chapters):
+        return self.__flatten_list_elements([chapter.get("lessons") for chapter in chapters])
+
+
+class VueSchoolLessonIE(VueSchoolBaseIE):
+    IE_DESC = 'VueSchool Lesson'
+    _VALID_URL = r'https?://vueschool\.io/lessons/(?P<id>[a-z0-9\-]+)'
+    _TESTS = [{
+        'url': 'https://vueschool.io/lessons/getting-started-with-vue-js-and-the-composition-api',
+        'info_dict': {
+            'id': '665769592',
+            'title': 'Getting Started with Vue.js and the Composition API',
+            'series_id': 'vue-js-fundamentals-with-the-composition-api',
+            'series': 'Vue.js 3 Fundamentals with the Composition API',
+            'chapter_number': 1,
+            'chapter': 'Getting Started with Vue.js and the Composition API',
+            'episode_number': 1,
+            'episode': 'Getting Started with Vue.js and the Composition API',
+            'duration': 238,
+            'uploader_id': str,
+            'uploader': str,
+            'uploader_url': str,
+            'description': str,
+            'thumbnail': str
+        },
+        'params': {
+            'skip_download': True,
+            'ignore_no_formats_error': True,
+        },
+    }]
+
+    def _real_extract(self, url):
+        lesson_slug = self._match_id(url)
+        webpage = self._download_webpage(url, video_id=lesson_slug)
+
+        """
+         Attempt to access content behind paywall
+        """
+        try:
+            (_, vimeo_wrapper_element) = get_element_text_and_html_by_tag("vimeo-player", webpage)
+            vimeo_player_wrapper_dict = extract_attributes(vimeo_wrapper_element)
+            vimeo_id = vimeo_player_wrapper_dict.get(":id")
+        except compat_HTMLParseError:
+            return self.raise_login_required()
```

Comment:
```suggestion
            self.raise_login_required(msg='This video is only available for paid users', method='cookies')
```

You should change the `msg` part if it's wrong